### ISSUE TITLE
BUG: fastparquet interface fails on load with non-unique index and CoW

### ISSUE THREAD

**Issue by max3-2** [pandas-dev/pandas#57673]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
from pathlib import Path
import pandas as pd
import numpy as np

# Set the random seed for reproducibility
np.random.seed(42)

# Create random data
data = {
    'Integer_Column': np.random.randint(0, 100, size=30),
    'Float_Column': np.random.uniform(0, 1, size=30),
    'String_Column1': [np.random.choice(['apple', 'banana', 'orange']) for _ in range(30)],
    'String_Column2': [np.random.choice(['cat', 'dog', 'rabbit']) for _ in range(30)],
    'String_Column3': [np.random.choice(['red', 'blue', 'green']) for _ in range(30)]
}
target = Path('sample_file_test.parquet')

# Create DataFrame
df = pd.DataFrame(data)

# Make index non-unique!
df.index = np.random.randint(0, 5, size=30)
df = df.sort_index()

# Write the parquet
df.to_parquet(target, engine='fastparquet')

# Load with CoW, error
pd.options.mode.copy_on_write = True
df_loaded = pd.read_parquet(target)

# Load with warn - no warning AND data loads
pd.options.mode.copy_on_write = 'warn'
df_loaded = pd.read_parquet(target)
```

### Issue Description

When writing a dataframe using the `fastparquet` engine of pandas that contains **non-unique indexing**, reloading said dataframe fails.  While the error is, in core, thrown by `fastparquet` I report it here since its directly related to the upcoming changes in `pandas`

### Expected Behavior

See example above - it is expected that the data loads back in as it is with the old settings of `CoW`

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : f538741432edf55c6b9fb5d0d496d2dd1d7c2457
python                : 3.11.4.final.0
python-bits           : 64
OS                    : Windows
OS-release            : 10
Version               : 10.0.19045
machine               : AMD64
processor             : Intel64 Family 6 Model 140 Stepping 1, GenuineIntel
byteorder             : little
LC_ALL                : None
LANG                  : en
LOCALE                : de_DE.cp1252

pandas                : 2.2.0
numpy                 : 1.24.4
pytz                  : 2023.3.post1
dateutil              : 2.8.2
setuptools            : 68.2.2
pip                   : 23.2.1
Cython                : 3.0.0
pytest                : None
hypothesis            : None
sphinx                : 7.2.6
blosc                 : None
feather               : None
xlsxwriter            : 3.1.9
lxml.etree            : 4.9.3
html5lib              : None
pymysql               : None
psycopg2              : None
jinja2                : 3.1.2
IPython               : 8.17.2
pandas_datareader     : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.2
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : 2024.2.0
fsspec                : 2023.10.0
gcsfs                 : None
matplotlib            : 3.8.1
numba                 : 0.57.1
numexpr               : 2.8.7
odfpy                 : None
openpyxl              : 3.1.2
pandas_gbq            : None
pyarrow               : None
pyreadstat            : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : 1.11.1
sqlalchemy            : None
tables                : 3.8.0
tabulate              : None
xarray                : None
xlrd                  : 2.0.1
zstandard             : 0.22.0
tzdata                : 2023.3
qtpy                  : 2.4.1
pyqt5                 : None
</details>

**Comment by jorisvandenbossche** [comment]
@max3-2 Thanks for the report. It's hard to say without seeing a traceback of the error you get, but I assume this is an issue in fastparquet: https://github.com/dask/fastparquet/issues/919 (essentially, fastparquet is not yet fully compatible with the new CoW mode)

**Comment by jorisvandenbossche** [comment]
The error is indeed "ValueError: assignment destination is read-only", so I am going to close this in favor of the issue on the fastparquet side (as this is something that needs to fixed there): https://github.com/dask/fastparquet/issues/919

**Comment by max3-2** [comment]
Yeah my bad I skipped the traceback in favor of the repoducer. But yes, the error reported is the one I was referring to and since it only occurs with non-unique indexing I placed it here.
Thanks for referring to the other issue!
